test/performance/benchmarks/rollout-probe/continuous/main.go

```
@@ -0,0 +1,131 @@
+	// across the different workload types.
+	t, ok := targets[*target]
+	if !ok {
+		fatalf("Unrecognized target: %s", *target)
+	}
+
+	// Make sure the target is ready before sending the large amount of requests.
+	if err := performance.ProbeTargetTillReady(t.target.URL, *duration); err != nil {
+		fatalf("Failed to get target ready for attacking: %v", err)
+	}
+
+	// Set up the threshold analyzers for the selected benchmark.  This will
+	// cause Mako/Quickstore to analyze the results we are storing and flag
+	// things that are outside of expected bounds.
+	q.Input.ThresholdInputs = append(q.Input.ThresholdInputs, t.analyzers...)
+
+	// Send 2000 QPS (2 per ms) for the given duration with a 30s request timeout.
+	rate := vegeta.Rate{Freq: 3, Per: time.Millisecond}
+	targeter := vegeta.NewStaticTargeter(t.target)
+	attacker := vegeta.NewAttacker(vegeta.Timeout(30 * time.Second))
+
+	// Create a new aggregateResult to accumulate the results.
+	ar := metrics.NewAggregateResult(int(duration.Seconds()))
+
+	// Start the attack!
+	results := attacker.Attack(targeter, rate, *duration, "load-test")
+LOOP:
+	for {
+		select {
+		case <-ctx.Done():
+			// If we timeout or the pod gets shutdown via SIGTERM then start to
+			// clean thing up.
+			break LOOP
+
+		case res, ok := <-results:
+			if !ok {
+				// Once we have read all of the request results, break out of
+				// our loop.
+				break LOOP
+			}
+			// Handle the result for this request
+			metrics.HandleResult(q, *res, t.stat, ar)
+		}
+	}
+
+	// Walk over our accumulated per-second error rates and report them as
+	// sample points.  The key is seconds since the Unix epoch, and the value
+	// is the number of errors observed in that second.
+	for ts, count := range ar.ErrorRates {
+		q.AddSamplePoint(mako.XTime(time.Unix(ts, 0)), map[string]float64{
+			t.estat: float64(count),
+		})
+	}
+
+	// Commit data to Mako and handle the result.
+	if err := mc.StoreAndHandleResult(); err != nil {
```

In case you missed it, if you want to report the error to Slack autoscaling channel, you need to configure it in `performance-tests.sh`

---

not yet.
I want to build the full one first.